### PR TITLE
currency of a single ticket could be updated now

### LINKED ISSUE #1913: Not able to update the free ticket's currency, created when a new conference is created
**I'm submitting a ..**
- [X] Bug Report
- [ ] Feature Request

**Current behavior:**

The free ticket created at time of conference creation has default currency as USD. 
This ticket's currency is not updating, even when this is the only conference ticket available.

ticket-bug-1

**Expected correct behavior:**

The ticket's currency should update whenever it is the only ticket of the conference
**Steps to reproduce:**

1) Make a new conference (or there should be a single ticket)
2) Try to update the free ticket's currency

### PR DESCRIPTION
**Pull request by ViditChitkara**
closes #1913

**Checklist**

- [x] I have read the [Contribution & Best practices Guide](https://github.com/openSUSE/osem/blob/master/CONTRIBUTING.md).
- [x] My branch is up-to-date with the upstream `master` branch.
- [x] The tests pass locally with my changes.
- [ ] I have added tests that prove my fix is effective or that my feature works(if appropriate).
- [ ] I have added necessary documentation (if appropriate).

### CODE REVIEW

File: app/models/ticket.rb [hunk 1]
```diff
@@ -72,8 +72,11 @@ def tickets_sold
   private
 
   def tickets_of_conference_have_same_currency
-    unless Ticket.where(conference_id: conference_id).all?{|t| t.price_currency == price_currency }
-      errors.add(:price_currency, 'is different from the existing tickets of this conference.')
+    tickets = Ticket.where(conference_id: conference_id)
+    unless tickets.count.zero? || (tickets.count == 1 && self == tickets.first)
```
Comment (differentreality) [inline app/models/ticket.rb]: How about instead of this unless statement you use
`return true if condition`

Comment (ViditChitkara) [reply app/models/ticket.rb]: That's a nice idea, but I guess return if condition should be fine (for line 76). There is no need to return 'true'. What do you think?

**Comment by differentreality** [comment]
@ViditChitkara are you going to take care of this?

**Comment by ViditChitkara** [comment]
Yes, I'll take care of this!! Just pushing the changes

**Comment by ViditChitkara** [comment]
@differentreality I have made the changes!! Could you please have a look at this?

**Comment by differentreality** [comment]
This looks good @ViditChitkara 
There is an extra empty file probably by mistake

I would like to add tests that would have caught the bug, how are your rspec skills? :)

**Comment by ViditChitkara** [comment]
Still learning about rspec. Will give it a go!!😃😃

**Comment by ViditChitkara** [comment]
@differentreality I have made the changes! Please have a look!

**Comment by ViditChitkara** [comment]
@differentreality Any changes required for this??

**Comment by ViditChitkara** [comment]
@differentreality rebased!!

File: spec/models/ticket_spec.rb [hunk 2]
```diff
@@ -177,4 +178,25 @@
       end
     end
   end
+
+  describe 'currency updation' do
```
Comment (differentreality) [inline spec/models/ticket_spec.rb]: This should say `currency update`

**Comment by hennevogel** [comment]
I guess we can live with the wrong work in the `spec`, merging.

**Pull request merged**